### ISSUE TITLE
Can not obtain accuracy mentioned in keras documentation for some models

### ISSUE THREAD

**Issue by sabujlaskar** [calebrob6/imagenet_validation#3]
Hi,
Thanks for providing such an awesome repository to test inference with imagenet and Keras. However, when applying your approach to different models, I didn't find accuracy similar to the mentioned in Keras [documentation.](https://keras.io/api/applications/) Specifically, I haven't found mentioned accuracy for the following datasets:

- Resnet50V2(Got- 70%(top1), 90%(top5), Mentioned in Keras doc-76%(top1), 93%(top5))
- Resnet101V2
- Resnet152V2
- Xception
- NasnetLarge
- InceptionV3
- InceptionResnetV2
- EfficientNetB0-B7

For all the models, I used model-specific preprocessing. For example, when in the code you used preprocessing from vggnet for the VGGNet model. Similarly, I used preprocessing of densenet for DenseNet models and so on. For those models, which has different input shape, I used an Input tensor of size (224, 224, 3) when model initialization. 

Can you kindly explain what might cause this issue?

**Comment by calebrob6** [comment]
I'm guessing some models use different input sizes (i.e. not 224 * 224) but
am not sure. Do the keras docs have default shapes for these models?

**Comment by sabujlaskar** [comment]
Resnet50V2, Resnet101V2, Resnet152V2 all these models have (224, 224, 3) input shape. Still, they are lagging by 5%-6% accuracy.
Other models have different shapes

1. Xception(299, 299, 3)
2. NasnetLarge(331, 331, 3)
3. InceptionV3(299, 299, 3)
4. InceptionResnetV2(299, 299, 3)
5. Efficientnet B1-B7 also has different input shapes

If input shapes are a probable issue, then can you please suggest preprocessing techniques during the data preparation part to adapt with different input shapes?

**Comment by sabujlaskar** [comment]
I fixed this issue by changing the input dimension while preparing the dataset. In the code, resize shape was 256 before center cropping for the image while the input dimension was 224. I kept the same ratio while varying the input dimensions. For example, I resized images to (342, 342) before center cropping while the input dimension was 299. By applying this technique, I found mentioned accuracies from the Keras docs except for the resnetV2 models. Thanks a lot for this awesome work.
